lib/mcts/node.rb

```diff
@@ -40,43 +40,32 @@ def expand
     end
 
     def rollout
-      playout = Playout.new(@game_state)
-      playout.play
+      state = @game_state.dup
+      state.set_move(state.generate_move) until state.finished?
+      state.winner
     end
 
-    def won
-      @visits += 1
-      @wins += 1
-    end
-
-    def lost
-      @visits += 1
-    end
-
-    def backpropagate(won)
+    def backpropagate(winner:)
       node = self
-      node.update_won won
-      until node.root? do
-        won = !won # switching players perspective
-        node = node.parent
-        node.update_won(won)
-      end
+      node.update_stats(winner: winner) until (node = node.parent).nil?
```

Comment:
Any reasoning against using the `root?` method?

Reply:
Sounds good to me.